src/nvim/os/time.c

```diff
@@ -46,26 +47,45 @@ void os_delay(uint64_t milliseconds, bool ignoreinput)
     }
     LOOP_PROCESS_EVENTS_UNTIL(&main_loop, NULL, (int)milliseconds, got_int);
   } else {
-    os_microdelay(milliseconds * 1000);
+    os_microdelay(milliseconds * 1000u, ignoreinput);
```

Comment:
@oni-link If I'm not mistaken, the old code was a regression from Vim. This PR makes it match `os_unix.c:mch_delay()`.